File: c++/0049-group-anagrams/main.cpp
#include <cstdlib>
#include <iostream>
#include <string>
#include <vector>

#include "solution.cpp"

using namespace std;

void printAns(vector<vector<string>>& ans) {
    cout << "[";

    for (int i = 0; i < ans.size(); ++i) {
        for (int j = 0; j < ans[i].size(); ++j) {
            if (j == 0) {
                cout << "[";
            }

            cout << ans[i][j];

            if (j != ans[i].size() - 1) {
                cout << ", ";
            } else {
                cout << "]";
            }
        }

        if (i != ans.size() - 1) {
            cout << ", ";
        }
    }

    cout << "]";
}

int main() {
    Solution solution;

    vector<string> strs1 = {"eat", "tea", "tan", "ate", "nat", "bat"};
    vector<vector<string>> ans1 = solution.groupAnagrams(strs1);
    cout << "[[bat], [nat, tan], [ate, eat, tea]]: ";
    printAns(ans1);
    cout << endl;

    vector<string> strs2 = {""};
    vector<vector<string>> ans2 = solution.groupAnagrams(strs2);
    cout << "[[]]: ";
    printAns(ans2);
    cout << endl;

    vector<string> strs3 = {"a"};
    vector<vector<string>> ans3 = solution.groupAnagrams(strs3);
    cout << "[[a]]: ";
    printAns(ans3);
    cout << endl;

    return EXIT_SUCCESS;
}

File: java/0049-group-anagrams/Main.java
public class Main {
    public static void main(String[] args) {
        Solution solution = new Solution();

        String ans1 = "[[bat], [nat, tan],[ate, eat, tea]]";
        String[] strs1 = new String[] { "eat", "tea", "tan", "ate", "nat", "bat" };
        System.out.println(ans1 + ": " + solution.groupAnagrams(strs1));

        String ans2 = "[[]]";
        String[] strs2 = new String[] { "" };
        System.out.println(ans2 + ": " + solution.groupAnagrams(strs2));

        String ans3 = "[[a]]";
        String[] strs3 = new String[] { "a" };
        System.out.println(ans3 + ": " + solution.groupAnagrams(strs3));
    }
}
